packages/sdk-core/src/modules/database/BacktraceDatabase.ts

```diff
@@ -188,30 +196,37 @@ export class BacktraceDatabase implements BacktraceModule {
     /**
      * Sends all records available in the database to Backtrace and removes them
      * no matter if the submission process was successful or not.
+     * @param abortSignal optional abort signal to cancel sending requests
      */
-    public async flush() {
+    public async flush(abortSignal?: AbortSignal) {
         const start = TimeHelper.now();
-        await this.send();
+        await this.send(abortSignal);
         const records = this.get().filter((n) => n.timestamp <= start);
         for (const record of records) {
             this.remove(record);
         }
```

Comment:
should we remove only records that were actually sent, if `flush` was aborted?

Reply:
No, we shouldn't. The expectation is to clear the database via flush. Send will also take care of cleaning them if we successfully send them.

Reply:
if someone would use abort signal in this context the expectation would be to stop the execution. I think you're correct. I added an if statement to the loop.